clio-client/src/main/scala/org/broadinstitute/clio/client/commands/Commands.scala

```diff
@@ -2,39 +2,104 @@ package org.broadinstitute.clio.client.commands
 
 import akka.http.scaladsl.model.HttpResponse
 import com.typesafe.scalalogging.LazyLogging
+import io.circe.parser._
 import org.broadinstitute.clio.client.ClioClientConfig
+import org.broadinstitute.clio.client.parser.BaseArgs
+import org.broadinstitute.clio.client.util.IoUtil
+import org.broadinstitute.clio.client.webclient.ClientAutoDerivation._
+import org.broadinstitute.clio.client.webclient.ClioWebClient
+import org.broadinstitute.clio.transfer.model.{
+  TransferReadGroupV1Key,
+  TransferReadGroupV1Metadata,
+  TransferReadGroupV1QueryInput
+}
+import org.broadinstitute.clio.util.model.Location
 
 import scala.concurrent.{Await, ExecutionContext, Future}
 
 object Commands {
-  val addReadGroupBam = "addReadGroupBam"
-  val queryReadGroupBam = "queryReadGroupBam"
-}
+  sealed abstract class Command(
+    val commandName: String
+  )(implicit ec: ExecutionContext)
+      extends LazyLogging {
+    def execute(clioWebClient: ClioWebClient, config: BaseArgs): Boolean
 
-abstract class Command(name: String)(implicit ec: ExecutionContext)
-    extends LazyLogging {
+    def checkResponse(responseFuture: Future[HttpResponse]): Boolean = {
+      Await.result(
+        responseFuture.map[Boolean] { response =>
+          val isSuccess = response.status.isSuccess()
 
-  def execute: Boolean
+          if (isSuccess) {
+            logger.info(
+              s"Successfully completed command $commandName." +
+                s" Response code: ${response.status}"
+            )
+          } else {
+            logger.error(
+              s"Error executing command $commandName." +
+                s" Response code: ${response.status}"
+            )
+          }
+          isSuccess
+        },
+        ClioClientConfig.responseTimeout
+      )
+    }
+  }
+  case class AddReadGroupBam()(implicit ec: ExecutionContext)
```

Comment:
Rather than bundling the `execute` functionality up inside the case classes, could we:
1. Make `Command` be an enum using enumeratum
2. Move the logic for executing commands inside `CommandDispatch`

I think it'd be an improvement because:
1. `Command` is sealed, so as we move forward with adding commands this file is going to get pretty big if we keep the `execute` logic in here too; moving the logic into `CommandDispatch` would let us break the functionality up into traits / typeclasses that the dispatcher uses
2. With the sub-commands as enums, we won't need to do the weird `apply()` calls in the parser to get their names